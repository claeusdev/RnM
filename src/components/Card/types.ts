import { Character } from "types";

export interface CharacterProps {
    character: Character;
  }