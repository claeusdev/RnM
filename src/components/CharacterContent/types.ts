import { Character, Location } from "types";


export interface CharacterContentProps {
  gender: string;
  location: Location;
  origin: { name: string };
  species: string;
}
