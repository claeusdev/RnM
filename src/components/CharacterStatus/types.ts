export interface CharacterStatusProps {
  status: string;
}
