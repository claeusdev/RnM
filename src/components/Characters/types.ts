import { Character } from "types";

export type CharactersProps = {
    page: string
    results: Character[]
}