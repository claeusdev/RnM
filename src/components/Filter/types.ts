export interface FilterProps {
    updatePageNumber: (pageNumber: number) => void;
}
