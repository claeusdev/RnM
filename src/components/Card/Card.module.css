.avatar {
  --avatar-size: 7.5rem;
  --avatar-bg: var(-border-color-light);
  position: relative;
  width: var(--avatar-size);
  height: var(--avatar-size);
  font-size: calc(var(--avatar-size) / 2.8571428572);
  font-weight: 500;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  color: #626976;
  text-align: center;
  text-transform: uppercase;
  vertical-align: bottom;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: var(--avatar-bg) no-repeat center/cover;
  border-radius: 4px;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
}

.icon {
  width: 1.25rem;
  height: 1.25rem;
  font-size: 1.25rem;
  vertical-align: bottom;
  stroke-width: 1.5;
  margin-right: 0.3rem;
}

.hr {
  display: flex;
  align-items: center;
  margin: 2rem 0;
  font-size: 1.25rem;
  font-weight: 600;
  text-transform: uppercase;
  letter-spacing: 0.04em;
  line-height: 1.6;
  color: #626976;
  height: 1px;
}
