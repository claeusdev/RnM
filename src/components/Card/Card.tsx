import CharacterStatus from 'components/CharacterStatus';
import { Episode } from 'types';
import styles from './Card.module.css';
import { CharacterProps } from './types';

export default function CharacterCard({ character, episodes }: CharacterProps) {
  return (
    <>
      <div className="page-header">
        <div className="row align-items-center">
          <div className="col-auto">
            <span
              className={styles.avatar}
              style={{
                backgroundImage: `url(${character.image})`,
              }}
            ></span>
          </div>
          <div className="col">
            <h2 className="page-title">{character.name}</h2>
            <div className="page-subtitle">
              <div className="row">
                <CharacterStatus status={character.status} />
                <div className="col-auto">
                  <span className="text-reset">{character.species}</span>
                </div>
                <div className="col-auto">
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className="icon icon-tabler icon-tabler-man"
                    width="24"
                    height="24"
                    viewBox="0 0 24 24"
                    strokeWidth="2"
                    stroke="currentColor"
                    fill="none"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  >
                    <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
                    <circle cx="12" cy="5" r="2"></circle>
                    <path d="M10 22v-5l-1 -1v-4a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v4l-1 1v5"></path>
                  </svg>
                  <span className="text-reset">{character.gender}</span>
                </div>
                <div className="col-auto">
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className={styles.icon}
                    width="24"
                    height="24"
                    viewBox="0 0 24 24"
                    strokeWidth="2"
                    stroke="currentColor"
                    fill="none"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  >
                    <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
                    <circle cx="12" cy="12" r="3"></circle>
                    <circle cx="12" cy="12" r="8"></circle>
                    <line x1="12" y1="2" x2="12" y2="4"></line>
                    <line x1="12" y1="20" x2="12" y2="22"></line>
                    <line x1="20" y1="12" x2="22" y2="12"></line>
                    <line x1="2" y1="12" x2="4" y2="12"></line>
                  </svg>
                  <span className="text-reset">{character.location.name}</span>
                </div>
                <div className="col-auto">
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className={styles.icon}
                    width="24"
                    height="24"
                    viewBox="0 0 24 24"
                    strokeWidth="2"
                    stroke="currentColor"
                    fill="none"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  >
                    <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
                    <circle cx="12" cy="11" r="3"></circle>
                    <path d="M17.657 16.657l-4.243 4.243a2 2 0 0 1 -2.827 0l-4.244 -4.243a8 8 0 1 1 11.314 0z"></path>
                  </svg>
                  <span className="text-reset">{character.origin.name}</span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className={styles.hr}>Episodes</div>
      <ul>
        {episodes.map((episode: Episode) => (
          <li key={episode.id}>{episode.name}</li>
        ))}
      </ul>
    </>
  );
}
