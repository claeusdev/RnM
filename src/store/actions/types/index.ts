export enum ActionTypes {
    LOAD_CHARACTERS = "LOAD_CHARACTERS",
    LOAD_CHARACTERS_SUCCESS = "LOAD_CHARACTERS_SUCCESS",
    LOAD_CHARACTERS_ERROR = "LOAD_CHARACTERS_ERROR",
    FILTER_CHARACTERS = "FILTER_CHARACTERS",
    FILTER_CHARACTERS_SUCCESS = "FILTER_CHARACTERS_SUCCESS",
    FILTER_CHARACTERS_ERROR = "FILTER_CHARACTERS_ERROR",
    LOAD_CHARACTER = "LOAD_CHARACTER",
    LOAD_CHARACTER_SUCCESS = "LOAD_CHARACTER_SUCCESS",
    LOAD_CHARACTER_ERROR = "LOAD_CHARACTER_ERROR",
    SET_EPISODES = "SET_EPISODES"
}