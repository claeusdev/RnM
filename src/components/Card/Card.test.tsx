import { render } from "@testing-library/react";
import Card from "./Card";
import { testCharacter } from "types/testData";

test("renders learn react link", () => {
  const component = render(<Card character={testCharacter} />);
  expect(component).toMatchSnapshot();
});
