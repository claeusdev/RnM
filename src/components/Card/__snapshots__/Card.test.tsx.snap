// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-header"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-auto"
          >
            <span
              class="avatar"
              style="background-image: url(https://rickandmortyapi.com/api/character/avatar/1.jpeg);"
            />
          </div>
          <div
            class="col"
          >
            <h2
              class="page-title"
            >
              Rick Sanchez
            </h2>
            <div
              class="page-subtitle"
            >
              <div
                class="row"
              >
                <div
                  class="col-auto text-success"
                >
                  Alive
                </div>
                <div
                  class="col-auto"
                >
                  <span
                    class="text-reset"
                  >
                    Human
                  </span>
                </div>
                <div
                  class="col-auto"
                >
                  <svg
                    class="icon icon-tabler icon-tabler-man"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                      stroke="none"
                    />
                    <circle
                      cx="12"
                      cy="5"
                      r="2"
                    />
                    <path
                      d="M10 22v-5l-1 -1v-4a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v4l-1 1v5"
                    />
                  </svg>
                  <span
                    class="text-reset"
                  >
                    Male
                  </span>
                </div>
                <div
                  class="col-auto"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                      stroke="none"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="3"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="8"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="2"
                      y2="4"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="20"
                      y2="22"
                    />
                    <line
                      x1="20"
                      x2="22"
                      y1="12"
                      y2="12"
                    />
                    <line
                      x1="2"
                      x2="4"
                      y1="12"
                      y2="12"
                    />
                  </svg>
                  <span
                    class="text-reset"
                  >
                    Citadel of Ricks
                  </span>
                </div>
                <div
                  class="col-auto"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                      stroke="none"
                    />
                    <circle
                      cx="12"
                      cy="11"
                      r="3"
                    />
                    <path
                      d="M17.657 16.657l-4.243 4.243a2 2 0 0 1 -2.827 0l-4.244 -4.243a8 8 0 1 1 11.314 0z"
                    />
                  </svg>
                  <span
                    class="text-reset"
                  >
                    Earth (C-137)
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hr"
      >
        Episodes
      </div>
      <ul />
    </div>
  </body>,
  "container": <div>
    <div
      class="page-header"
    >
      <div
        class="row align-items-center"
      >
        <div
          class="col-auto"
        >
          <span
            class="avatar"
            style="background-image: url(https://rickandmortyapi.com/api/character/avatar/1.jpeg);"
          />
        </div>
        <div
          class="col"
        >
          <h2
            class="page-title"
          >
            Rick Sanchez
          </h2>
          <div
            class="page-subtitle"
          >
            <div
              class="row"
            >
              <div
                class="col-auto text-success"
              >
                Alive
              </div>
              <div
                class="col-auto"
              >
                <span
                  class="text-reset"
                >
                  Human
                </span>
              </div>
              <div
                class="col-auto"
              >
                <svg
                  class="icon icon-tabler icon-tabler-man"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <circle
                    cx="12"
                    cy="5"
                    r="2"
                  />
                  <path
                    d="M10 22v-5l-1 -1v-4a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v4l-1 1v5"
                  />
                </svg>
                <span
                  class="text-reset"
                >
                  Male
                </span>
              </div>
              <div
                class="col-auto"
              >
                <svg
                  class="icon"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="8"
                  />
                  <line
                    x1="12"
                    x2="12"
                    y1="2"
                    y2="4"
                  />
                  <line
                    x1="12"
                    x2="12"
                    y1="20"
                    y2="22"
                  />
                  <line
                    x1="20"
                    x2="22"
                    y1="12"
                    y2="12"
                  />
                  <line
                    x1="2"
                    x2="4"
                    y1="12"
                    y2="12"
                  />
                </svg>
                <span
                  class="text-reset"
                >
                  Citadel of Ricks
                </span>
              </div>
              <div
                class="col-auto"
              >
                <svg
                  class="icon"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <circle
                    cx="12"
                    cy="11"
                    r="3"
                  />
                  <path
                    d="M17.657 16.657l-4.243 4.243a2 2 0 0 1 -2.827 0l-4.244 -4.243a8 8 0 1 1 11.314 0z"
                  />
                </svg>
                <span
                  class="text-reset"
                >
                  Earth (C-137)
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="hr"
    >
      Episodes
    </div>
    <ul />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
